src/js/components/PlacementConstraintsWidget.js

```diff
@@ -0,0 +1,238 @@
+import React, { Component } from "react";
```

Comment:
This file is direct copy-paste from `PlacementSection`. The intention is this widget will be used in `PlacementSection` as done above, and `SchemaField` as done below.